packages/contracts/contracts/Artonomous.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity ^0.4.23;
+
+import "./ArtonomousStaking.sol";
+import "./ArtonomousArtPieceToken.sol";
+
+contract Artonomous {
+
+    event ArtonomousAuctionStarted(uint indexed blockNumber, string generatorHashUsed);
+    event ArtonomousArtClaimed(address indexed claimant, uint indexed blockNumber, string generatorHashUsed);
+
+    struct Auction {
+        uint blockNumber;
+        string generatorHash;
+        uint endTime;
+    }
+
+    ArtonomousStaking public artonomousStaking;
+    ArtonomousArtPieceToken public pieceToken;
+    uint public AUCTION_LENGTH = 86400; // 24 hours
+    Auction currentAuction;
+
+    constructor(address stakingAddr) public {
+        artonomousStaking = ArtonomousStaking(stakingAddr);
+        pieceToken = new ArtonomousArtPieceToken("artonomous-token", "ARTO");
+    }
+
+    function startAuction() external {
+        require(currentAuction.blockNumber == 0);
+        string memory currentGeneratorHash = artonomousStaking.currentGeneratorHash();
+        currentAuction = Auction({
+            blockNumber: block.number,
+            generatorHash: currentGeneratorHash,
```

Comment:
Not 100% sure, but it might be cheaper to mint the token on auction start and then transfer on claim. That way we can avoid storing the `generatorHash` here.

Reply:
good idea, I'll make that change. In fact, I think that "start auction" won't even be an externally accessible function. We'll just start an auction on contract deploy and start another one every time a piece is collected. That way, the beneficiary pays all the gas and we don't have to rely on someone paying gas to start auctions benevolently.